src/test/java/net/datafaker/script/ProvidersDocsGenerator.java

```diff
@@ -0,0 +1,188 @@
+        final File file = new File(filePath);
+
+        Optional<CommentsCollection> commentsCollection = parser.parse(file).getCommentsCollection();
+
+        if (!commentsCollection.isPresent()) {
+            fakersWithoutSinceTag.add(filePath);
+            return "";
+        }
+
+        Optional<JavadocComment> javadocComments = commentsCollection.get().getJavadocComments()
+            .stream()
+            .findFirst();
+
+        if (!javadocComments.isPresent()) {
+            fakersWithoutSinceTag.add(filePath);
+            return "";
+        }
+
+        String comment = javadocComments.get().getContent();
+        boolean isComment = comment.contains("@since");
+
+        if (!isComment) {
+            fakersWithoutSinceTag.add(filePath);
+        }
+
+        return comment;
+    }
+    public String generateColumn(String name, char padSymbol, int length) {
+        if (name.length() >= length) return name;
+        return padSymbol + name + Strings.repeat(String.valueOf(padSymbol), length - name.length() - 1);
+    }
+    /**
+     * Writes header and table header to new 'providers.md'
+     *
+     * @see #writeToFile(String)
+     */
+    private void constructHeaderInProvidersFile() {
+        final int sinceColLength = 7;
+        final String header = "# Fake Data Providers\n" + "\n" + "Datafaker comes with the following list of data providers:" + "\n";
+        final String tableHeader = "\n|" + generateColumn("Name", ' ', nameColLength) + "|" + generateColumn("Description", ' ', descColLength) + "|" + generateColumn("Since", ' ', sinceColLength) + "|\n";
+        final String tableHeaderDelimiter = "|" + generateColumn("", '-', nameColLength) + "|" + generateColumn("", '-', descColLength)  + "|" + generateColumn("", '-', sinceColLength) + "|";
+
+        try {
+            writeToFile(header);
+            writeToFile(tableHeader);
+            writeToFile(tableHeaderDelimiter);
+        } catch (IOException e) {
+            throw new RuntimeException("Error on 'providers.md' file creation", e);
+        }
+    }
+
+    private void writeProviderToTable(final String providerName, final String sinceTag) {
+        String nameEntry = "\n|" + generateColumn("[" + addSpaceBetweenNameOfProvider(providerName) + "]" +
+            "({{ datafaker.javadoc }}/" + providerName + ".html)", ' ', nameColLength) + "|";
+        final String providerDescriptionSectionEntry = Strings.repeat(" ", descColLength) ;
+        // Format of `@since tag` should be '#.#.#'
+        String providerSinceTagSectionEntry = "|" + " " + sinceTag + " " + "|";
+        try {
+            writeToFile(nameEntry + providerDescriptionSectionEntry + providerSinceTagSectionEntry);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * When generating new file, the old one should be removed, because new data will be written in existing one.
```

Comment:
It seems the comments does not match functionality: it does not remove old file and always adds.
Moreover it seems that each time this method is called it opens file again and again for every row.
It would be better to open/recreate a file once somewhere before the cycle in `generateProvidersDocs` and then just write into it

Reply:
I mean when you run the entire program more than once, it just will append the entire file structure in the existing one

Reply:
> the old one should be removed
does it imply manually?

Reply:
yes

Reply:
ok, just created a pr to your branch https://github.com/panilya/datafaker/pull/1
it allows to forget about such manual work